File: ClientApp/src/app/models/Article.ts
import { Category } from ".";
import { Section } from "./Section";
import { User } from "./User";
import { Wiki } from "./Wiki";

export class Article {
  id: number;
  title: string;
  abstract: string;
  wiki: Wiki;
  wikiID: number;
  sections: Section[];
  categories: Category[];
  creator: User;
  creationDate: string;
  modificationDate: string;
  active: boolean


  constructor(object: any) {
    if(object.id != null){
      this.id = object.id;
    }
    this.title = object?.title || '';
    this.abstract = object?.abstract || '';
    this.active = object?.active || false;
    if(object.wiki != null){
      this.wiki = new Wiki(object.wiki);
      this.wikiID = this.wiki.id;
    }
    else if(object.wikiID != null){
      this.wikiID = object?.wikiID;
    }
    if(object.owner != null){
      this.creator = new User(object.owner);
    }
    if(object.creationDate != null){
      this.creationDate = object.creationDate
    }
    if(object.modificationDate != null){
      this.modificationDate = object.modificationDate
    }
    this.sections = [];
    object?.sections?.forEach((obj:string) => { this.sections.push(new Section(obj)); })||[];

    this.categories = [];
    object?.categories?.forEach((obj:string) => { this.categories.push(new Category(obj)); })||[];
  }


}

File: Server/Entities/Article/Models/Article.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace OpenWiki.Server.Entities {
    public class Article {
        [Key]
        public long ID { get; set; }
        public string Title { get; set; }
        public string Abstract { get; set; }
        public Wiki Wiki { get; set; }
        public ICollection<Section> Sections { get; set; }
        [ForeignKey("CreatorId")]
        public ApplicationUser Creator { get; set; }
        [ForeignKey("ModifierId")]
        public ApplicationUser Modifier { get; set; }
        public DateTime CreationDate { get; set; }
        public DateTime ModificationDate { get; set; }
    }
}
